Documentation/Reference.md

```diff
@@ -61,51 +61,117 @@ Where `foo` is an executable and `bar.a` a static library.
 
 ## Package Manifest File Format Reference
 
-Instructions for how to build a package are provided by a manifest file, called `Package.swift`. You can customize this file to declare build targets or dependencies, include or exclude source files, and specify build configurations for the module or individual files.
+Instructions for how to build a package are provided by the `Package.swift` manifest file. `Package.swift` is a Swift file that declaratively configures a Package using types defined in the `PackageDescription` module. It currently has following properties:
 
-Here's an example of a `Package.swift` file:
+### name
+
+Defines the name of the package and is the minimal requirement for a manifest to be valid. When the sources are located directly under `Sources/` directory, there is only one module and the module name is same as the package name.
+
+```swift
+import PackageDescription
+
+let package = Package(
+    name: "FooBar"
+)
+```
+
+### targets
+
+Targets property is used when you have more than one module in your package and want to declare inter-dependency between them.
+
+Note: It is also possible to declare target dependencies between a test and regular module.
 
 ```swift
 import PackageDescription
 
 let package = Package(
     name: "Hello",
-    dependencies: [
-        .Package(url: "ssh://git@example.com/Greeter.git", versions: Version(1,0,0)..<Version(2,0,0)),
+    targets: [
+        Target(name: "TestSupport"),
+        Target(name: "Bar", dependencies: ["Foo"]),
+        Target(name: "BarTests", dependencies: ["Bar", "TestSupport"]),
     ]
 )
 ```
 
-A `Package.swift` file is a Swift file that declaratively configures a Package using types defined in the `PackageDescription` module. This manifest declares a dependency on an external package: `Greeter`.
+Here `Foo`, `Bar` and `TestSupport` are modules present under `Sources/` directory and `BarTests` is a test module under `Tests/` directory.
+`Bar` module can import `Foo` module and `BarTests` module can import `Foo`, `Bar` and `TestSupport` modules.
 
-If your package contains multiple targets that depend on each other you will need to specify their interdependencies. Here is an example:
+### dependencies
+
+This is the list of packages that the current package depends on. You can specify url (or local path) to any valid swift package placed under
+version control which is tagged appropriately.
 
 ```swift
 import PackageDescription
 
 let package = Package(
     name: "Example",
-    targets: [
-        Target(
-            name: "top",
-            dependencies: [.Target(name: "bottom")]),
-        Target(
-            name: "bottom")
+    dependencies: [
+        .Package(url: "ssh://git@example.com/Greeter.git", versions: Version(1,0,0)..<Version(2,0,0)),
+        .Package(url: "../StringExtensions", majorVersion: 1),
+        .Package(url: "https://github.com/MyAwesomePackage", majorVersion: 1, minor: 4),
     ]
 )
 ```
```

Comment:
There are several initializers for the `Package.Dependency` type, we should have a complete reference here.

We might want to veer a little more towards documenting at the Swift level, including the Swift types for things. For example, we could organize the documentation as something like:

```
Package(name: String, targets: [Target], dependencies: [Package.Dependency])

 - name: The name of the package.
 - targets: Additional information on each target. See <reference>.
```

Just like developers would expect from any other API reference.